cmd/tether/attach.go

```diff
@@ -179,131 +168,147 @@ func (t *attachServerSSH) stop() {
 func (t *attachServerSSH) run() error {
 	defer trace.End(trace.Begin("main attach server loop"))
 
-	var sConn *ssh.ServerConn
+	var serverConn *ssh.ServerConn
 	var chans <-chan ssh.NewChannel
 	var reqs <-chan *ssh.Request
 	var err error
 
-	// keep waiting for the connection to establish
-	for t.enabled && sConn == nil {
-		conn := t.conn
-		if conn == nil {
-			// Stop has probably been called as t.conn is set in Start and should
-			// never be nil otherwise
-			err := fmt.Errorf("connection provided for backchannel is nil")
-			log.Debug(err.Error())
-			return err
-		}
+	for t.enabled {
+		// keep waiting for the connection to establish
+		for serverConn == nil {
+			// tests are passing their own connections so do not create connections when testing is set
+			if !t.testing {
```

Comment:
Can we add a mock for this, or even add it into backchannel?

backchannel was originally opening the connection as well as running the handshake, but when this was split out as an extension that was moved into Start.
If we need to reinitialize the connection in this way for serial then putting it into backchannel is a much cleaner approach.